sdmetrics/reports/base_report.py

```diff
@@ -80,10 +80,10 @@ def _validate(self, real_data, synthetic_data, metadata):
                 The metadata of the table.
         """
         self._validate_data_format(real_data, synthetic_data)
-        self._validate_metadata_matches_data(real_data, synthetic_data, metadata)
+        if self.__class__.__name__ == 'DiagnosticReport':
+            return
 
-    def _handle_results(self, verbose):
-        raise NotImplementedError
+        self._validate_metadata_matches_data(real_data, synthetic_data, metadata)
```

Comment:
I think we should just override the `_validate_metadata_matches_data` method in the `DiagnosticReport` to be a no-op instead. Main reason being in case any naming changes. I don't expect that it ever will, but I think we should avoid any introspection when possible.

Reply:
I see, done in [d58d19d](https://github.com/sdv-dev/SDMetrics/pull/524/commits/d58d19d64b6c69ae3303192289685e0d96856d44)